### DISCUSSION TITLE
[Enhancement] Blazor in MAUI Preference

### DISCUSSION THREAD

**Discussion by AhmedAbuelnour** [dotnet/maui#179] [General]
## Summary

I want to share my preference about using Blazor in MAUI,

- be able to easy Bind events to Methods and passing whatever number of parameters as we do in Blazor, without the need for behaviors (XAML way for handling it)

- be able to sperate the logic and design as in Blazor's components 

- be able to register pages for navigation with directives like ``` @page ```

- be able to access instances from Dependency Container by [Inject] Attribute like in Blazor

- be able to access any object in my "Code-behind" not like in XAML when defining a DataTemplate i can only bind to itemsource's item object, for example if i want to bind to a command, i want it to be more straightforward.

All the listed above are just examples for what i really want to see in MAUI, there still more to mention but they are the core features that will make it more elegant to use Blazor as an option for Web-Developers who used to work with Blazor

**Comment by GeraudFabien** [comment]
Seem like a santa claus list.

> be able to access any object in my "Code-behind" not like in XAML when defining a DataTemplate i can only bind to itemsource's item object, for example if i want to bind to a command, i want it to be more straightforward.

Not sure what you means by that. Can you explain this part ?
If you mean to access you're XAML control instance in Controller/ViewModel/.... You already can do it. But it's not recommended since it's tightly couple you're view and Controller/ViewModel/.... In this case it didn't go well with 'separate logic and design'.
Or maybe do you mean the somethinks a bit different like MVU ?
   

>be able to easy Bind events to Methods and passing whatever number of parameters as we do in Blazor, without the need for behaviors (XAML way for handling it)

The real problem is that not everythinks is bindable. Otherwise you wouldn't have to make Behavior.
   
> be able to sperate the logic and design as in Blazor's components

it's the main goal of MVC/MVP/PM/MVVM/MVU. Ensuring standard and uncouple way for the "vue" to talk with rest of the application. That what MVC bring. MVVM offer decoupling between Mode, View and Viewmodel themself by using DuckTyping (Data Transfert Object ex: Model, ViewModel), Event (Simple object related message ex: INotifyPropertyChanged, INotifyCollectionChanged) and messageBus (Complex message or related to global context between view and viewmodel ex: Open dialog, Toaster) to communicate between the three.
What can be added to the current Xamarin ? I don't see why Blazor since in it the two are by default very heavly couple. Only the dependency injection help wich again have nothinks to do with blazor.


>be able to easy Bind events to Methods and passing whatever number of parameters as we do in Blazor, without the need for behaviors (XAML way for handling it)
>
>be able to sperate the logic and design as in Blazor's components
>
>be able to register pages for navigation with directives like @page
>
>be able to access instances from Dependency Container by [Inject] Attribute like in Blazor
   
In a nutshell you want https://github.com/xamarin/MobileBlazorBindings
